tests/ui/branches_sharing_code/false_positives.rs

```diff
@@ -0,0 +1,48 @@
+#![allow(dead_code)]
+#![deny(clippy::if_same_then_else, clippy::branches_sharing_code)]
+
+// ##################################
+// # Issue clippy#7369
+// ##################################
+#[derive(Debug)]
+pub struct FooBar {
+    foo: Vec<u32>,
+}
+
+impl FooBar {
+    pub fn bar(&mut self) {
+        if true {
+            self.foo.pop();
+        } else {
+            self.baz();
+
+            self.foo.pop();
+
+            self.baz()
+        }
+    }
+
+    fn baz(&mut self) {}
+}
+
+// #########################
+// # Issue 7452
+// #########################
+fn test() {
+    let v = vec![1, 2, 3, 4, 5, 6, 7, 8, 9, 0];
+    let mut counter = 0;
+    for i in v {
+        if counter == 0 {
+            counter += 1;
+            println!("first");
+        } else if counter == 1 {
+            counter += 1;
+            println!("second");
+        } else {
+            counter += 1;
+            println!("other: {}", i);
+        }
+    }
+}
```

Comment:
This is the issue that is not fixed yet. I found it reasonable to add a test for it (even if it fails) as it's also documented in the lint description

Reply:
I'm not sure we gain anything by adding this. Tests are valuable when they verify correct behavior.

Reply:
Yeah, I wasn't sure with this one that's, the reason why I pointed it out. I'll remove it from this PR and then squash the two commits :upside_down_face: